### PR TITLE
[PCM-754] Fix proxying disconnect event, add reconnect method

### PR DESCRIPTION
**Pull request by xdumaine**

### CODE REVIEW

**Comment by xdumaine** [comment]
@samueljun FYI

File: src/client-private.js [hunk 1]
```diff
@@ -183,7 +183,7 @@ function proxyStreamingClientEvents () {
 
   // other events
   on('error', this.emit.bind(this, 'error'));
-  on('disconnected', () => this.emit('disconnected', 'Streaming API connection disconnected'));
+  on('disconnect', () => this.emit('disconnected', 'Streaming API connection disconnected'));
```
Comment (xdumaine) [inline src/client-private.js]: This event was wrongly wired up before. It also was undocumented.

File: doc/documentation.md [hunk 2]
```diff
@@ -227,6 +230,10 @@ messages for the session manager
     - `Error error` - the error that occurred
     - `Object details` - the details about the error
 
+`sdk.on('disconnected', () => {})` - the underlying websocket connection has
```
Comment (ckbrewer33) [inline doc/documentation.md]: shouldn't that be `sdk.on('disconnect'` ?

Comment (jensengar) [reply doc/documentation.md]: no, the event coming from realtime is `disconnect`, the one that gets forwarded is `disconnected`.

Comment (ckbrewer33) [reply doc/documentation.md]: Yep, I mixed it up

Comment (bjdupuis) [reply doc/documentation.md]: Looks like `disconnected` is still being emitted.

Comment (xdumaine) [reply doc/documentation.md]: No, we're still going to emit `disconnected`, which is changed from the underlying websocket lib which emits `disconnect` instead.

**Comment by xdumaine** [comment]
Published as v1.0.6

**Pull request merged**